### ISSUE TITLE
Should S108 be detected in `tempfile.TemporaryDirectory(dir="/dev/shm")`?

### ISSUE THREAD

**Issue by mihaic** [astral-sh/ruff#6355]
Considering the following contents of `issue.py` (no other files, no `pyproject.toml`):
```python
import tempfile

with tempfile.TemporaryDirectory(dir="/dev/shm") as tmp:
    pass
```

S108 is detected:
```
ruff check --select S108 --isolated issue.py
issue.py:3:38: S108 Probable insecure usage of temporary file or directory: "/dev/shm"
Found 1 error.
```

Shouldn't the usage of `TemporaryDirectory` preclude S108?

```
ruff --version
ruff 0.0.282
```

**Comment by charliermarsh** [comment]
Yeah, it should probably be allowed.
